Browser

hasbel opened the pull request:
Add a browser-bundle package and an example of how to use it. Update docs accordingly.

The browser-bundle packages build to a 167kb javascript file that can be used as browser-side library and includes:
- core
- binding-http (client only)
- binding-websocket (client-only)

danielpeintner commented on README.md:
````diff
@@ -98,7 +134,24 @@ To reduce the size of the installation from about 800 MByte down to about 200 MB
+
+Node-wot can also be imported as browser-side library. To do so, include the following `script` tag in your html:
+```html
+<script src="https://cdn.jsdelivr.net/npm/@node-wot/browser-bundle@latest/dist/wot-bundle.min.js"></script>
````
I assume this CDN links works once the wot-bundle is pushed on NPM, Correct?
Currently it fails.

Existing packages like https://cdn.jsdelivr.net/npm/@node-wot/td-tools@latest/dist/td-tools.min.js work.

Shall we add a note to readers?

hasbel replied:
Yes, once it is on NPM it will work. This is actually the best approach in my opinion as the alternative would be to upload the compiled code on `/dist` to github and use the github link, which should be avoided, and it should not be a problem if a snapshot is released on NPM directly after this PR is merged.

I also think that it would be useful to have more SNAPSHOT releases available on NPM for node-wot, as currently anyone using NPM to get node-wot would be missing on all the changes made in the last 2 months.

The typescript-definitions for example are already on 0.7.0-SNAPSHOT-5.

danielpeintner commented on examples/browser/index.html:
```diff
@@ -0,0 +1,58 @@
+
+    <!-- defer to not block rendering -->
+    <script src="https://cdn.jsdelivr.net/npm/@json-editor/json-editor@latest/dist/jsoneditor.min.js" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/@node-wot/browser-bundle@latest/dist/wot-bundle.min.js" defer></script>
```
Hence, this fails also...

danielpeintner commented on examples/browser/index.html:
```diff
@@ -0,0 +1,58 @@
+    <!-- defer to not block rendering -->
+    <script src="https://cdn.jsdelivr.net/npm/@json-editor/json-editor@latest/dist/jsoneditor.min.js" defer></script>
+    <script src="https://cdn.jsdelivr.net/npm/@node-wot/browser-bundle@latest/dist/wot-bundle.min.js" defer></script>
+    <script src="index.js" defer></script>
```
When loading in FF66 I besides the failing resource I get some more warnings.

e.g., ReferenceError: JSONEditor is not defined

Maybe it would be good to add some try catch code to let the user know that somethings did not work as expected.

hasbel replied:
The first defer should be removed. Thanks I will fix this.

danielpeintner commented on README.md:
```diff
@@ -3,6 +3,15 @@ W3C Web of Things implementation on NodeJS
+- [Start with an example](#example)
+- [How to use the library](#api)
+
+<a name="license"/>
```
All the anchors, also the others are not needed and I would not add them because they are already there and automatically updated (see https://github.com/eclipse/thingweb.node-wot#prerequisites etc)

danielpeintner commented:
@hasbel thank you for your great work and efforts. It looks very promising. I added some comments (see above).

At the moment the HTML example is not working given that the bundle is not on NPM yet.
Shall we put in on some other resource until we are ready. 

The problem I see is that otherwise it might confuse people...

danielpeintner commented:
One more comment: we have a *running* thing instance on http://plugfest.thingweb.io:8083/TestThing.

We could say that instead of starting the counter example one can use that URL instead.
(see http://plugfest.thingweb.io:8088/test/fullscreen/default)

hasbel commented:
@danielpeintner Thanks a lot for the quick feedback! 

Regarding NPM, as I commented above, I think that it is the best approach when combined with a snapshot release. Having more snapshot releases of node-wot would be a good thing IMO.

Thanks for letting me know about the testThing. I replaced the counter and updated the table of content as you suggested.
